### DISCUSSION TITLE
Additional empty line in code block

### DISCUSSION THREAD

**Discussion by nomyfan** [remarkjs/remark#699] [Q&A] [answered]
There's always a new empty line(\n) in the end of the code block.

My markdown is below
````
hello world
~~~js
console.log('js')
~~~
````

The code above will get the `children` to be `["console.log('js')\n"]` which I think the `\n` should not be existed.
This is a [demo](https://codesandbox.io/s/jolly-moser-ckr4q?file=/src/index.js:275-314) to reproduce.

**Comment by wooorm** [comment, marked as answer]
That’s how it works in markdown: https://spec.commonmark.org/dingus/?text=%60%60%60x%0Ay%0A%60%60%60.

And why in the readme that line ending is removed: https://github.com/remarkjs/react-markdown#use-custom-components-syntax-highlight.

I think it could be an issuer with `react-syntax-highlighter`. You might want to raise it there.
